src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs

```diff
@@ -253,6 +253,10 @@ internal T GetValue<T>(string key, bool throwEx, out bool found)
                 if (long.TryParse(obj.ToString(), out long value))
                     return (T)(object)value;
             }
+            else if(typeof(T).IsEnum)
```

Comment:
I think enum should be lower on the stack, below ICollection<string>

Reply:
Fair enough considering that enums won't be used so often. Done.